lib/ansible/modules/cloud/amazon/iam.py

```diff
@@ -728,7 +728,7 @@ def main():
                 (user_groups, changed) = set_users_groups(
                     module, iam, name, groups, been_updated, new_name)
             module.exit_json(
-                user_meta=meta, groups=user_groups, keys=keys, changed=changed)
+                user_meta=meta, groups=user_groups, user_name=meta['created_user']['user_name'], keys=keys, changed=changed)
```

Comment:
It seems like it's already in user_meta key. Not sure whether it is needed.

Reply:
Not sure what you mean. The point of this change to make user_name available independent of action performed under the hood so plays that rely on `user_name` key don't fail on reruns (where user is already present).